tests/test_multipart.py

```diff
@@ -599,6 +599,18 @@ async def test_read_form(self, newline: Any) -> None:
             result = await obj.form()
         assert [("foo", "bar"), ("foo", "baz"), ("boo", "")] == result
 
+    async def test_read_form_invalid_utf8(self, newline: Any) -> None:
+        data = b"\xff%s--:--" % newline
+        with pytest.raises(ValueError):
+            with Stream(data) as stream:
+                obj = aiohttp.BodyPartReader(
+                    BOUNDARY,
+                    {CONTENT_TYPE: "application/x-www-form-urlencoded"},
+                    stream,
+                    _newline=newline,
+                )
+                await obj.form()
```

Comment:
What is actually supposed to raise a `ValueError`? This line? If yes, move the `pytest.raises()` CM here to isolate where we expect the exception to arise. Otherwise, the test will be misleading and will pass even if `aiohttp.BodyPartReader()` raises a `ValueError` not even calling `.form()`.